worlds/sc2/__init__.py

```diff
@@ -377,6 +343,8 @@ def resolve_count(count: Optional[int], max_count: int) -> int:
                 result[-1].flags |= ItemFilterFlags.Locked
```

Comment:
From what I've seen so far, "locked" and "necessary" are still unchanged (one task suggests renaming them from "locked", "necessary" -> "requested", "locked"). This means the user option is still hooked up to the flag that can be overridden, which isn't what we want, I think. So this should add the "necessary" flag (we can handle the rename later).

Reply:
This is a preparation step. Mentioned that in #309